### PR TITLE
Type Hints for aioredis

### PR DESCRIPTION
**Pull request by butuzov**
## What do these changes do?

+ Enabls type hints using `mypy`
+ Adds initial type checks for `errors.py` and `parser.py`

## Are there changes in behavior for the user?

None

## Related issue number
#555

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is `<Name> <Surname>`.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,6 +58,14 @@ jobs:
     - pip install -r tests/requirements.txt
     script:
     - make flake
+ - &MYPY
+    name: mypy
+    cache: false
+    python: "3.6"
```
Comment (webknjaz) [inline .travis.yml]:
```suggestion
    python: "3.7"
```

Comment (butuzov) [reply .travis.yml]: Makes total sense. Added 3.7 version for Travis-CI job.

File: aioredis/parser.py [hunk 2]
```diff
@@ -35,31 +36,33 @@ def gets(self):
...
-        self.replyError = replyError
-        self.encoding = encoding
+    def __init__(self, protocolError: Callable,
+                 replyError: Callable, encoding: Optional[str] ):
+
+        self.buf: bytearray = bytearray()
+        self.pos: int = 0
+        self.protocolError: Callable = protocolError
+        self.replyError: Callable = replyError
+        self.encoding: Optional[str] = encoding
```
Comment (popravich) [inline aioredis/parser.py]: This wont work in Python 3.5

Comment (popravich) [reply aioredis/parser.py]: I mean variable annotations

**Pull request merged**